graph/step.go

```diff
@@ -196,3 +201,49 @@ func (s *Step) UpdateBuildStepWithDefaults() {
 		s.Build = fmt.Sprintf("--isolation hyperv %s", s.Build)
 	}
 }
+
+// UseBuildCacheForBuildStep indicates if buildx needs to be used
+func (s *Step) UseBuildCacheForBuildStep() bool {
+	if s == nil {
+		return false
+	}
+
+	if s.EnableCache && s.CacheID == "" {
+		s.CacheID = "cache"
+	}
+
+	return s.IsBuildStep() && s.CacheID != ""
+}
+
+// GetCmdWithCacheFlags adds buildx cache parameters to the cmd
+func (s *Step) GetCmdWithCacheFlags() (string, error) {
+	result := ""
+
+	if s == nil {
+		return result, errors.New("step is null")
+	}
+
+	if s.Tags == nil || len(s.Tags) == 0 {
+		return result, errors.New("atleast one tag is required to use buildcache's cache registry")
+	}
+
+	buildImg := s.Tags[0]
+	repo, err := reference.Parse(buildImg)
+	if err != nil {
+		return result, errors.Wrap(err, "failed to parse the image name")
+	}
+
+	if named, ok := repo.(reference.Named); ok {
+		// also trim out any digests
+		named = reference.TrimNamed(named)
+		cacheImage, err := reference.WithTag(named, s.CacheID)
+		if err != nil {
+			return result, errors.Wrap(err, "failed to attach cache ID tag to the repo for buildcache")
+		}
+
+		// todo (sam) check if we need `--push` to be able to push the cache.
```

Comment:
`--push` is not used for pushing the cache. It is used to push the resulted image.

Basically, `buildx` uses a `buildkitd` container to build the image as the build driver since you did `buildx create`. The container is a different system that it does not share images with the docker daemon outside. As a resolution, `buildx build` has an option [-o, --output](https://github.com/docker/buildx/blob/master/README.md#-o---outputpath-typetypekeyvalue) to export the resulted image. Two notable extended options are 
- `--push Shorthand for --output=type=registry` push the image after building completion
- `--load Shorthand for --output=type=docker` export the image to docker daemon after building completion so that you can find the image using `docker images` later.

Reply:
You may take a look at `--load`. This option may be useful for Tasks.

Reply:
Thanks, seems like `--push` is what I am looking for. Please see the updated code. How would `--load` be helpful?